src/main/kotlin/uk/gov/dluhc/printapi/client/BankHolidayDataFileClient.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.dluhc.printapi.client
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.type.MapType
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.io.Resource
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import javax.annotation.PostConstruct
+
+@Component
+class BankHolidayDataFileClient(
+    @Value("classpath:bank-holidays.json") private val resource: Resource,
```

Comment:
I'm sure this works, but is this the right syntax for an `@Value` annotation ??
```
@Value("classpath:bank-holidays.json") 
```
As I understand it the annotation is used to resolve SPEL expressions, and 99.9% of the time used to get property values. Therefore I would expect this line to be something like
```
@Value("\${bank-holidays.data-file.location}") 
```
or similar. And then it would be defined in `application.properties` as
```
bank-holidays:
    data-file:
        location: classpath:bank-holidays.json
```
I suspect its working more by luck than judgement (on the basis that `classpath:bank-holidays.json` is being resolved as a string literal SPEL expression)

Reply:
Bear in mind it's a tactical solution so I was hoping not to modify application.yml, unless we have to.

I lifted this approach from the interweb somewhere, but happy to change it if necessary.

Reply:
It's a fair challenge as I know you'd normally use a `@Value` annotation as you said. I stole it from here: https://stackoverflow.com/a/58705928. I'll dig a bit deeper.